digiwf-engine/digiwf-engine-service/src/main/java/io/muenchendigital/digiwf/process/instance/infrastructure/repository/ProcessInstanceInfoRepository.java

```diff
@@ -23,4 +27,37 @@ public interface ProcessInstanceInfoRepository extends JpaRepository<ServiceInst
     List<ServiceInstanceEntity> findAllByInstanceIdIn(List<String> instanceIds);
 
     List<ServiceInstanceEntity> findByRemovalTimeBefore(Date referenceDate);
+
+
+    @Query(
+            value = "SELECT si FROM io.muenchendigital.digiwf.process.instance.infrastructure.entity.ServiceInstanceEntity si LEFT JOIN io.muenchendigital.digiwf.process.instance.infrastructure.entity.ServiceInstanceAuthorizationEntity sia ON sia.processInstanceId = si.instanceId WHERE sia.userId = :user",
+            countQuery = "SELECT count(si) FROM io.muenchendigital.digiwf.process.instance.infrastructure.entity.ServiceInstanceEntity si LEFT JOIN io.muenchendigital.digiwf.process.instance.infrastructure.entity.ServiceInstanceAuthorizationEntity sia ON sia.processInstanceId = si.instanceId WHERE sia.userId = :user"
+    )
+    Page<ServiceInstanceEntity> findAllByUserId(@Param("user") String userId, Pageable pageable);
+
+    @Query(
+            value = "SELECT si FROM io.muenchendigital.digiwf.process.instance.infrastructure.entity.ServiceInstanceEntity si LEFT JOIN io.muenchendigital.digiwf.process.instance.infrastructure.entity.ServiceInstanceAuthorizationEntity sia ON sia.processInstanceId = si.instanceId WHERE sia.userId = :user AND (" +
+                    "lower(si.id) LIKE concat('%', :search,'%')" +
+                    "OR lower(si.instanceId) LIKE concat('%', :search,'%')" +
+                    "OR lower(si.definitionName) LIKE concat('%', :search,'%')" +
+                    "OR lower(si.definitionKey) LIKE concat('%', :search,'%')" +
+                    "OR lower(si.description) LIKE concat('%', :search,'%')" +
+                    "OR lower(si.status) LIKE concat('%', :search,'%')" +
+                    "OR lower(si.statusKey) LIKE concat('%', :search,'%')" +
+                    ")",
+            countQuery = "SELECT count(si) FROM io.muenchendigital.digiwf.process.instance.infrastructure.entity.ServiceInstanceEntity si LEFT JOIN io.muenchendigital.digiwf.process.instance.infrastructure.entity.ServiceInstanceAuthorizationEntity sia ON sia.processInstanceId = si.instanceId WHERE sia.userId = :user AND (" +
+                    "lower(si.id) LIKE concat('%', :search,'%')" +
+                    "OR lower(si.instanceId) LIKE concat('%', :search,'%')" +
+                    "OR lower(si.definitionName) LIKE concat('%', :search,'%')" +
+                    "OR lower(si.definitionKey) LIKE concat('%', :search,'%')" +
+                    "OR lower(si.description) LIKE concat('%', :search,'%')" +
+                    "OR lower(si.status) LIKE concat('%', :search,'%')" +
+                    "OR lower(si.statusKey) LIKE concat('%', :search,'%')" +
+                    ")"
+    )
+    Page<ServiceInstanceEntity> searchAllByUserId(
+            @Param("search") String lowerQuery,
+            @Param("user") String userId,
+            Pageable pageable
+    );
```

Comment:
Genau solche Queries wollte ich eigentlich vermeiden, da nicht typsicher und ggfs. DB abhängig. Ich hatte das im Ref angesprochen und ihr meintet, dass es nicht nötig sei. Ist Criteria Query hier nicht besser?

Reply:
Nachdem die beiden Queries nicht mehr ganz trivial sind würde ich zumindest noch Javadoc hinzufügen

Reply:
Meiner meinung nach ist Criteria an dicher stelle auch nicht typsicher. am Ende generiert es ja auch "nur JPQL". Bedeutet wir haben mit Criteria die gleichen Probleme. Desweiteren suchen wir nur in Varchar Spalten. Somit sollte das DB unabhängig sein. 

Ich kann da noch Javadoc dran schreiben. Aber an sich ist das nur ein Join mit einer oder verknüpften suche. Ich schreibe es aber nochmal hin. :-)

Reply:
Haben im Daily drüber gesprochen. Passt für mich